src/util/key.rs

```diff
@@ -481,4 +500,53 @@ mod tests {
         assert_tokens(&pk_u.compact(), &[Token::BorrowedBytes(&PK_BYTES_U[..])]);
         assert_tokens(&pk_u.readable(), &[Token::BorrowedStr(PK_STR_U)]);
     }
+
+    fn random_key(mut seed: u8) -> PublicKey {
```

Comment:
What do people think about making this pub? I think all repositories that depend on rust-bitcoin have some function that does exactly this.

Reply:
Please NO! This is **not CSRNG**!

If people want it public, then please call it `not_crypto_safe_random_key_i_swear_to_use_it_for_testing_only`

Reply:
For what it's worth, nobody knows the discrete logs of the generated public keys, so it's probably really hard to actually cause a crypto break with this function.

But I'm curious what the purpose of it would be outside of unit tests :).

Reply:
Agree that is not CSRNG and it is not a good idea even if there is no secret key with it. It is already under mod tests, if we want it pub we can name it something else.